File: 515-find-largest-value-in-each-tree-row/find-largest-value-in-each-tree-row.cpp
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        std::vector<int> res; // Specify the type of elements as int
        if (root == nullptr) {
            return res;
        }

        std::vector<TreeNode*> queue; // Specify the type of elements as TreeNode*
        queue.push_back(root);

        while (!queue.empty()) {
            int maxVal = INT_MIN;
            int levelSize = queue.size();

            for (int i = 0; i < levelSize; i++) {
                TreeNode* node = queue.front();
                queue.erase(queue.begin());

                maxVal = std::max(maxVal, node->val);

                if (node->left != nullptr) {
                    queue.push_back(node->left);
                }

                if (node->right != nullptr) {
                    queue.push_back(node->right);
                }
            }

            res.push_back(maxVal);
        }

        return res;
    }
};

File: 515-find-largest-value-in-each-tree-row/find-largest-value-in-each-tree-row.js
var largestValues = function(root) {
    if (!root) {
        return [];
    }

    let queue = [root];
    let res = [];
    while (queue.length) {
        let levelSize = queue.length;
        let maxVal = -Infinity;
        for (let i = 0; i < levelSize; i++) {
            let node = queue.shift();
            maxVal = Math.max(maxVal, node.val);
            if (node.left) {
                queue.push(node.left);
            }
            if (node.right) {
                queue.push(node.right);
            }
        }
        res.push(maxVal);
    }
    return res;
};